src/azul/collections.py

```diff
@@ -271,3 +273,64 @@ def to_dict(self):
             k: v.to_dict() if isinstance(v, NestedDict) else v
             for k, v in self.items()
         }
+
+
+class OrderedSet(MutableSet[Any]):
+    """
+    A mutable set that maintains insertion order. It is not a sequence.
+
+    >>> OrderedSet() == set()
+    True
+    >>> 'a' in OrderedSet(['a', 'b'])
+    True
+    >>> OrderedSet(['a', 'b', 'c', 'b'])
+    OrderedSet(['a', 'b', 'c'])
+    """
+
+    def __init__(self, members: Iterable = (), /) -> None:
+        self.inner: dict = dict.fromkeys(members)
+
+    def __repr__(self) -> str:
+        if not self:
+            return f"{self.__class__.__name__}()"
+        return f"{self.__class__.__name__}({list(self)!r})"
+
+    def __iter__(self) -> Iterator:
+        return iter(self.inner.keys())
+
+    def __len__(self) -> int:
+        return len(self.inner)
+
+    def __eq__(self, member: object) -> bool:
+        return self.inner.keys() == member
+
+    def __contains__(self, member: object) -> bool:
+        return member in self.inner.keys()
+
+    def discard(self, member: Any) -> None:
+        """
+        >>> os = OrderedSet(['1', 'a', '2'])
+        >>> os.discard('a')
+        >>> os
+        OrderedSet(['1', '2'])
+        """
+        self.inner.pop(member)
```

Comment:
`discard` should not raise an exception if the member is not present in the set.
```suggestion
        self.inner.pop(member, None)
```

I would also add a doctest for this case.